File: motor3d 2.0/AngulosEuler.vb
Imports System.Math

Namespace Algebra
    Public Structure AngulosEuler
        Private mCabeceo As Single
        Private mAlabeo As Single
        Private mGuiñada As Single

        Public Property Cabeceo As Single
            Get
                Return mCabeceo
            End Get
            Set(value As Single)
                mCabeceo = value
            End Set
        End Property

        Public Property Alabeo As Single
            Get
                Return mAlabeo
            End Get
            Set(value As Single)
                mAlabeo = value
            End Set
        End Property

        Public Property Guiñada As Single
            Get
                Return mGuiñada
            End Get
            Set(value As Single)
                mGuiñada = value
            End Set
        End Property

        Public ReadOnly Property Cuaternion As Cuaternion
            Get
                Return ObtenerCuaternion(Me)
            End Get
        End Property

        Public Sub New(ByVal Cabeceo As Single, ByVal Alabeo As Single, ByVal Guiñada As Single)
            mCabeceo = Cabeceo
            mAlabeo = Alabeo
            mGuiñada = Guiñada
        End Sub

        Public Shared Function ObtenerCuaternion(ByVal Angulos As AngulosEuler) As Cuaternion
            Return New Cuaternion(Angulos.Cabeceo, Angulos.Alabeo, Angulos.Guiñada)
        End Function

        Public Overrides Function ToString() As String
            Return "{AngulosEuler: " & mCabeceo.ToString & "," & mAlabeo.ToString & "," & mGuiñada.ToString & "}"
        End Function
    End Structure
End Namespace

File: motor3d 2.0.ConvertedToC#/AngulosEuler.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Math;

namespace Motor3D.Algebra
{
	public struct AngulosEuler
	{
		private float mCabeceo;
		private float mAlabeo;

		private float mGuiñada;
		public float Cabeceo {
			get { return mCabeceo; }
			set { mCabeceo = value; }
		}

		public float Alabeo {
			get { return mAlabeo; }
			set { mAlabeo = value; }
		}

		public float Guiñada {
			get { return mGuiñada; }
			set { mGuiñada = value; }
		}

		public Cuaternion Cuaternion {
			get { return ObtenerCuaternion(this); }
		}

		public AngulosEuler(float Cabeceo, float Alabeo, float Guiñada)
		{
			mCabeceo = Cabeceo;
			mAlabeo = Alabeo;
			mGuiñada = Guiñada;
		}

		public static Cuaternion ObtenerCuaternion(AngulosEuler Angulos)
		{
			return new Cuaternion(Angulos.Cabeceo, Angulos.Alabeo, Angulos.Guiñada);
		}

		public override string ToString()
		{
			return "{AngulosEuler: " + mCabeceo.ToString() + "," + mAlabeo.ToString() + "," + mGuiñada.ToString() + "}";
		}
	}
}

